test/include/outputs/TestAutosaveTimedCheckpoint.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "Checkpoint.h"
+
+class TestAutosaveTimedCheckpoint : public Checkpoint
```

Comment:
Wait, do we need this object? I can see your comment about testing the autosave rather than a regular checkpoint, but I don't believe you need this. Checkpoint is doing both, we should be able to test both with that object.

Reply:
Combining the response with the bottom question with this comment -- the reason this test object exists is to test that the system checkpoint (the one created by the signal handler) is able to output the timed checkpoint we are doing now. Since we decided against the shortcut syntax, that means there's normally no way to modify the amount of time it takes for the system checkpoint to output the timed output we are performing here.

So, instead, we have this test object that we can make operate exactly like a system checkpoint -- we set the `TestAutosaveTimedCheckpoint` object to have the same `_is_autosave` flag as the system checkpoint, and then change the interval to 1 second so we can test the system checkpoint is outputting correctly. This is the easiest way to verify the automatically created checkpoint is working with the wall time output correctly.

I don't know if there's a good way to avoid the test object here to accomplish this. Obviously open to suggestions if you have any, but that's the justification for its existence.

Reply:
OK fair, but then wouldn't I expect to see one of your tests use `type = Checkpoint` (i.e. an explicit Checkpoint instance that does autosaving too)? Maybe I'm still missing something.

Reply:
The `TestAutosaveTimedCheckpoint` is just a derived Checkpoint object so there's no difference between it and a normal checkpoint if you don't pass the `test_system=true`